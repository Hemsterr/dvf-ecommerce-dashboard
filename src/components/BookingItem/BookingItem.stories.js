// Libs
import React from 'react'
import { storiesOf } from '@storybook/react'

// Components
import BookingItem from './index'

storiesOf('BookingItem', module).add('Default', () => (
  <BookingItem title="I would like to fit myself using Hemster's Fitting Kit" />
))
