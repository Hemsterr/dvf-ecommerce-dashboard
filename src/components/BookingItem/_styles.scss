.bookingItem {
  width: pxToRem(350px);
  height: pxToRem(400px);
  display: flex;
  justify-content: center;
  align-content: center;
  align-items: center;
  flex-wrap: wrap;
  background-size: cover;
  cursor: pointer;

  &__small {
    width: pxToRem(150px);
    height: pxToRem(200px);

     & .bookingItem__title {
      display: none;
     }
  }

  &__icon {
    width: pxToRem(75px);
    height: pxToRem(75px);
    display: block;
    background: url('../../assets/icons/fitting-icon.png') no-repeat;
  }

  &__fit-stylist {
    background-image: url('../../assets/images/fitting-virtually.png');

    & span {
      background-position: 0 pxToRem(-450px);
    }
  }

  &__fitting-kit {
    background-image: url('../../assets/images/fitting-kit.png');

    & span {
      background-position: 0 pxToRem(-300px);
    }
  }

  &__in-home-stylist {
    background-image: url('../../assets/images/fitting-stylist.png');

    & span {
      background-position: 0 pxToRem(-375px);
    }

    & p {
      padding: pxToRem(40px) pxToRem(85px) 0;
    }
  }

  &__title {
    @include text-size(pxToRem(15px), pxToRem(1px), pxToRem(30px));
    color: $alabaster;
    font-family: $font-base-futura;
    padding: pxToRem(40px) pxToRem(40px) 0;
  }
}

@media only screen and (max-width: $screen-lg) {
  .bookingItem  {
    width: pxToRem(280px);
    height: pxToRem(380px);

    &__small {
      width: pxToRem(150px);
      height: pxToRem(190px);
    }
  }
}

@media only screen and (max-width: $screen-md) {
  .bookingItem {
    width: 100%;
    margin-bottom: pxToRem(10px);

    &__title, &__in-home-stylist p {
      padding: pxToRem(30px) pxToRem(20px) 0;
    }

    &__fit-stylist {
      background-image: url('../../assets/images/fitting-virtually-mobile.png');
    }

    &__fitting-kit {
      background-image: url('../../assets/images/fitting-kit-mobile.png');
    }

    &__in-home-stylist {
      background-image: url('../../assets/images/fitting-stylist.png');
    }

    &__small {
      & .bookingItem__title {
        display: block;
      }
    }
  }
}

@media only screen and (max-width: $screen-xs) {
  .bookingItem  {
    height: pxToRem(150px);
    flex-wrap: nowrap;
    align-items: center;

    &__icon {
      width: pxToRem(40px);
      height: pxToRem(40px);
      margin: 0 pxToRem(25px);
    }

    &__title {
      padding: 0 pxToRem(40px) 0 0 ;
      @include text-size(pxToRem(12px), pxToRem(1px), pxToRem(25px));
    }

    &__fit-stylist {
      & span {
        background-position: 0 pxToRem(-845px);
      }
    }

    &__fitting-kit {
      & span {
        background-position: 0 pxToRem(-765px);
      }
    }

    &__in-home-stylist {
      & span {
        background-position: 0 pxToRem(-805px);
      }

      & p {
        padding: 0 pxToRem(35px) 0 0 ;
      }
    }
  }
}

@media only screen and (max-width: $screen-tn) {
  .bookingItem  {
    height: pxToRem(125px);

    &__icon {
      margin: 0 pxToRem(15px);
    }

    &__title {
      @include text-size(pxToRem(10px), pxToRem(1px), pxToRem(20px));
    }
  }
}
