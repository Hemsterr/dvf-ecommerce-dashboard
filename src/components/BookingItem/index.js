// @flow
import React from 'react'

type Props = {
  type: string,
  handleSelectOption: Function,
  isSmall: boolean,
  description: string,
}

class BookingItem extends React.PureComponent<Props> {
  static defaultProps = {
    description: '',
    type: 'fitting-kit',
    handleSelectOption: () => {},
    isSmall: false,
  }

  handleSelectOption = () => {
    const { handleSelectOption, type } = this.props
    handleSelectOption(type)
  }

  render() {
    const { type, description, isSmall } = this.props

    return (
      <div
        className={
          isSmall
            ? `bookingItem bookingItem__${type} bookingItem__small`
            : `bookingItem bookingItem__${type}`
        }
        onClick={this.handleSelectOption}
        role="presentation"
      >
        <div>
          <span className="bookingItem__icon" />
        </div>
        {!!description && <p className="bookingItem__title">{description}</p>}
      </div>
    )
  }
}

export default BookingItem
