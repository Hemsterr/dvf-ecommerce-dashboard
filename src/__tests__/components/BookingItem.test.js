// Libs
import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

// Component
import BookingItem from '../../components/BookingItem'

const mockProps = {
  description: "I would like to fit myself using Hemster's Fitting Kit",
  type: 'fit-stylist',
  handleSelectOption: jest.fn(),
}

const wrapper = (props = {}) => {
  const component = shallow(<BookingItem {...props} />)

  return component
}

describe('BookingItem component', () => {
  // Snapshot
  test('should render correctly BookingItem component default props', () => {
    const component = wrapper()
    expect(shallowToJson(component)).toMatchSnapshot()
    expect(BookingItem.defaultProps.description).toEqual('')
    expect(BookingItem.defaultProps.type).toEqual('fitting-kit')
    expect(BookingItem.defaultProps.handleSelectOption()).toBeUndefined()
  })

  // Testing props
  test('should be rendered with props', () => {
    const wrapperWithProps = wrapper(mockProps)
    expect(wrapperWithProps.find('.bookingItem__fit-stylist').length).toEqual(1)
    expect(
      wrapperWithProps.find('.bookingItem__title').props().children
    ).toEqual(mockProps.description)
  })

  test('should be rendered with small component', () => {
    const wrapperWithProps = wrapper({ ...mockProps, isSmall: true })
    expect(wrapperWithProps.find('.bookingItem__small').length).toEqual(1)
  })
})
