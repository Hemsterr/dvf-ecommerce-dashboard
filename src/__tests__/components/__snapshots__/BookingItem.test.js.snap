// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingItem component should render correctly BookingItem component default props 1`] = `
<div
  className="bookingItem bookingItem__fitting-kit"
  onClick={[Function]}
  role="presentation"
>
  <div>
    <span
      className="bookingItem__icon"
    />
  </div>
  <p
    className="bookingItem__title"
  />
</div>
`;
